### PR TITLE
updated models and implemented createInitialTest method in LearnEngin…

### PR DESCRIPTION
**Pull request by farhadkzm**
@ghahramani in model haro bebin doroste. yeki az method haye LearnEngine interface ro ham implement kardam.

### CODE REVIEW

File: app/src/main/java/com/smart/learning/service/learn/StupidLearnEngine.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+            }
+        }
+        while (initialSize != questions.size()); //if something is added, then there is a chance of adding more on the next iteration
+
+        return questions;
+    }
+
+    private Map<Integer, List<Content>> getContentsForDifficultyRange(Subject subject, Range<Integer> difficultyRange) {
+        List<Content> contentsOfSubject = new LinkedList<>();//todo fill this
```
Comment (farhadkzm) [inline app/src/main/java/com/smart/learning/service/learn/StupidLearnEngine.java:52]: @ghahramani  This needs to be implemented. ???

File: app/src/main/java/com/smart/learning/domain/learn/Content.java [hunk 2]
```diff
@@ -29,7 +32,18 @@
     /**
      * assigned tags by system or teachers
      */
-    private Set<Tag> tags;
+    private List<Tag> tags = new LinkedList<>();
```
Comment (ghahramani) [inline app/src/main/java/com/smart/learning/domain/learn/Content.java:35]: Farhad @DBRef injaro yadet rafte bezari

File: app/src/main/java/com/smart/learning/domain/learn/Subject.java [hunk 3]
```diff
@@ -16,9 +17,12 @@
     private String name;
 
     //student can go further and customize sub-subjects they want to learn
-    private Set<Subject> children;
+    @DBRef
+    private List<Subject> children;
 
-    private Set<Tag> tags;
+    private List<Tag> tags;
```
Comment (ghahramani) [inline app/src/main/java/com/smart/learning/domain/learn/Subject.java:23]: Injam hamintor @DBRef

File: app/src/main/java/com/smart/learning/service/learn/StupidLearnEngine.java (hunk 1 above)
Comment (ghahramani) [reply app/src/main/java/com/smart/learning/service/learn/StupidLearnEngine.java:52]: Yes, I think we need to get contents from service layer when we decided what kind of data we want to have for MVP

**Pull request closed without merging**